phpunit/class-block-library-comment-template-test.php

```diff
@@ -222,4 +222,52 @@ function test_rendering_comment_template_nested() {
 			'<ol ><li><div class="wp-block-comment-author-name">Test</div><div class="wp-block-comment-content">Hello world</div><ol><li><div class="wp-block-comment-author-name">Test</div><div class="wp-block-comment-content">Hello world</div><ol><li><div class="wp-block-comment-author-name">Test</div><div class="wp-block-comment-content">Hello world</div></li></ol></li></ol></li></ol>'
 		);
 	}
+	/**
+	 * Test that both "Older Comments" and "Newer Comments" are displayed in the correct order
+	 * inside the Comment Query Loop when we enable pagination on Discussion Settings.
+	 * In order to do that, it should exist a query var 'cpage' set with the $comment_args['paged'] value.
+	 */
```

Comment:
TBH without knowing much about `build_comment_query_vars_from_block`, pagination settings, and the `cpage` query arg and `paged` value, this is a lot to swallow 😅 It's not obvious to me how all these things are related to the correct order of "Older Comments" and "Newer Comments". 

The test may have its value in covering the underlying implementation to ensure the desired behavior, but maybe it'd actually make sense to still add the e2e test to cover that behavior?

Reply:
See https://github.com/WordPress/gutenberg/pull/39502/files#r831004637